cmd/kubeadm/app/cmd/phases/workflow/runner.go

```diff
@@ -0,0 +1,382 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workflow
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+// phaseSeparator defines the separator to be used when concatenating nested
+// phase names
+const phaseSeparator = "/"
+
+// RunnerOptions defines the options supported during the execution of a
+// kubeadm composable workflows
+type RunnerOptions struct {
+	// FilterPhases defines the list of phases to be executed (if empty, all).
+	FilterPhases []string
+
+	// SkipPhases defines the list of phases to be excluded by execution (if empty, none).
+	SkipPhases []string
+}
+
+// RunData defines the data shared among all the phases included in the workflow, that is any type.
+type RunData = interface{}
```

Comment:
What is the benefit of the type alias here?
 
versus type definition like this 

```
type RunData interface{}
```

Reply:
@chuckha 
I simply found ugly using interface{}, and thus created an alias for it.